### PR TITLE
Welcome Page

### PR DESCRIPTION
**Pull request by winder**
Feature #1025. A startup page designed to introduce new users to UGS. Most of the UI code (Tabs, show next time, tab-banner) are a taken from the netbeans welcome component with extra features removed. Then I created the tab content.

screen shot 2018-06-24 at 4 47 59 pm
screen shot 2018-06-24 at 4 48 08 pm
screen shot 2018-06-24 at 4 48 15 pm

### CODE REVIEW

**Comment by winder** [comment]
Here is the text if anyone has suggestions that would make it better:

**Getting Started**
>   private static String line1 = "Welcome to Universal Gcode Sender! If this is your first time using UGS thanks for giving it a try. This window is here to help you get started, once you're comfortable with the basic sending functionality please explore the other tabs to learn about some of the other features that UGS has to offer.";
> 
>   private static String openLine = "Open your Gcode file from the 'File' menu. Once you load a file there are a number of things you can do with it: inspect the toolpaths from the Visualizer, edit them by selecting edit from the 'File' menu, see the number of rows in the program in the UGS status area. UGS is a great way to preview your toolpaths even if you aren't ready to run the program yet!";
> 
>   private static String connectLine1 = "When you are ready to connect to your machine, the serial connection can be configured in the toolbar at the top of the program. Choose your firmware from the combo box and configure the serial options accordingly. If you haven't configured your controller yet you may be interested in the configuration wizard found in the 'Machine' menu.";
>   private static String connectLine2 = "Once connected you can start using UGS to control your machine. Some good places to start are the Jog Controller, manually sending commands in the console, or working with the coordinate system in the DRO. More information about these features can be found in the other tabs and at http://winder.github.io/ugs_website/";
> 
>   private static String runLine = "Once you have loaded a file and established a connection, the program may be run. These icons in the toolbar can be used to stream, pause, and stop the program.";
> 

**Features**
> new Feature(null, "Plugin Support!", "Most plugins can be found in the 'Window > Plugins' menu. New plugins are added all the time, see the what's new section for the latest features!", "com/willwinder/ugp/welcome/resources/features/new_plugins.png"),
>
> new Feature(null, "Setup wizard", "Configure and calibrate your hardware with the Setup wizard. It will help you to setup and test limit switches, step length, homing and soft limits. Start the wizard from the machine menu.", "com/willwinder/ugp/welcome/resources/features/setup_wizard.png"),
>
> new Feature(null, "Digital Read-Out (DRO)", "The DRO is your first stop for the current machine status. It tells you the work/machine coordinates, machine/spindle speeds, gcode state and more! It can be used to reset individual axes by clicking on the axis label, and has dynamic work position controls by clicking on the coordinate numbers.", "com/willwinder/ugp/welcome/resources/features/dro.png"),
>
> new Feature(null, "Jog Controller", "The Jog Controller is your primary tool for manually controlling your machines location. It has the option of using separate step sizes for XY and Z axes in addition to a click-and-hold mode for continuous jogging.", "com/willwinder/ugp/welcome/resources/features/jog_controller.png"),
>
> new Feature(null, "Custom Macros", "User defined macros can be configured in the preferences menu. A variety of convenient substitutions are available to help create perfectly tailored time savers for your workflow. See the help menu for more details.", "com/willwinder/ugp/welcome/resources/features/custom_macros.png"),
>
> new Feature(null, "Configurable Keybindings", "Nearly every feature in UGS can have a configurable keybinding. From the preferences menu open the keybinding section for a complete list of actions which can be configured (including your custom macros!).", "com/willwinder/ugp/welcome/resources/features/keybinding.png"),
>
> new Feature(null, "Probe Module", "The probe module can be opened from the plugins menu, it provides first class support for common probe devices. Some of the supported routines include standard Z-Depth probing and 3-Axis corner probing.", "com/willwinder/ugp/welcome/resources/features/probe_module.png"),
>
> new Feature(null, "Workflow helper", "Manage complex projects with multiple files using the Workflow helper. It will keep track of which files that has been run and remind you which tool to use for the next file.", "com/willwinder/ugp/welcome/resources/features/workflow.png")
>

File: ugs-core/src/com/willwinder/universalgcodesender/utils/Settings.java [hunk 1]
```diff
@@ -142,11 +149,37 @@ public String getLastOpenedFilename() {
...
+    public void setLastOpenedFilename(String absolutePath) {
+        Path p = Paths.get(absolutePath).toAbsolutePath();
+        this.fileName = p.toString();
+        updateRecentFiles(p.toString());
+        updateRecentDirectory(p.getParent().toString());
         changed();
     }
 
+    public Collection<String> getRecentFiles() {
+      return Collections.unmodifiableCollection(fileHistory);
```
Comment (breiler) [inline ugs-core/src/com/willwinder/universalgcodesender/utils/Settings.java]: Nice

File: ugs-core/src/com/willwinder/universalgcodesender/utils/Settings.java [hunk 2]
```diff
@@ -24,16 +24,21 @@ This file is part of Universal Gcode Sender (UGS).
...
 import org.apache.commons.lang3.StringUtils;
 
 public class Settings {
     // Transient, don't serialize or deserialize.
     transient private SettingChangeListener listener = null;
+    transient public static int HISTORY_SIZE = 20;
 
     private String firmwareVersion = "GRBL";
     private String fileName = System.getProperty("user.home");
+    private ArrayDeque<String> fileHistory = new ArrayDeque<>();
```
Comment (breiler) [inline ugs-core/src/com/willwinder/universalgcodesender/utils/Settings.java]: Really just a detail... =)

I usually want the member or a variable to have the most common interface specified which could make things simpler when for example refactoring:
```private Deque<String> fileHistory = new ArrayDeque<>();``

File: ugs-platform/ugs-platform-welcome-page/pom.xml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    </dependency>  
+    <dependency> 
+      <groupId>org.netbeans.api</groupId>  
+      <artifactId>org-openide-loaders</artifactId>  
+      <version>${netbeans.version}</version> 
+    </dependency>  
+    <dependency> 
+      <groupId>org.netbeans.api</groupId>  
+      <artifactId>org-openide-modules</artifactId>  
+      <version>RELEASE82</version>
```
Comment (breiler) [inline ugs-platform/ugs-platform-welcome-page/pom.xml]: I would like this to be `${netbeans.version}`instead

**Pull request merged**